app/models/test_track/application_identity.rb

```diff
@@ -12,7 +12,7 @@ def app_name
   end
 
   def identity
-    Identity.new(app_name)
+    @identity = @identity&.app_name == app_name ? @identity : Identity.new(app_name)
```

Comment:
```suggestion
    @identity ||= Identity.new(app_name)
```

If we're worried about `app_name` changing, we could make it so that it raises an exception if you try to reassign `TestTrack.app_name=` after a value has already been assigned... 🤔

Reply:
So, this is what I originally had, but then memoizing the value broke a test that we had where the `TestTrack.app_name` gets set in a `before` block. Basically as your thing is written, after memoizing, resetting `TestTrack.app_name` has no effect, so the test that verifies what happens when the `app_name` is not set fails (it expects an error but gets none because `app_name` was set in a previously run test).

Any thoughts as to how I can write the test such that that's not an issue? Because if that's reasonably possible, then yes this is also what I'd prefer!

Reply:
For the context of a test, we could:

```ruby
allow(TestTrack).to receive(:app_name).with(...)
```

Reply:
OK cool got it so the problem was that the identity ivar was persisting across specs so I set it up so that it's cleared out before all specs. Made `app_name` private again, and now just using `@identity ||= ...` 👍🏻